src/com/xilinx/rapidwright/design/compare/DesignComparator.java

```diff
@@ -0,0 +1,417 @@
+
+        for (Entry<String, Net> e : goldSiteWireMap.entrySet()) {
+            Net testNet = testSiteWireMap.remove(e.getKey());
+            if (testNet == null) {
+                addDiff(DesignDiffType.SITEWIRE_NET_MISSING, e.getKey(), null, gold, " should be Net " + e.getValue());
+                continue;
+            }
+            if (!e.getValue().getName().equals(testNet.getName())) {
+                addDiff(DesignDiffType.SITEWIRE_NET_NAME, e.getKey(), testNet, gold , "");
+            }
+        }
+        for (Entry<String, Net> e : testSiteWireMap.entrySet()) {
+            Net extraNet = e.getValue();
+            addDiff(DesignDiffType.SITEWIRE_NET_EXTRA, null, e.getKey(), test, " extra Net " + extraNet);
+        }
+        
+        // Active SitePIPs
+        Map<BEL, SitePIP> goldSitePIPs = getSitePIPMap(gold);
+        Map<BEL, SitePIP> testSitePIPs = getSitePIPMap(test);
+        
+        for (Entry<BEL, SitePIP> e : goldSitePIPs.entrySet()) {
+            SitePIP testPIP = testSitePIPs.remove(e.getKey());
+            if (testPIP == null) {
+                addDiff(DesignDiffType.SITEPIP_MISSING, e.getValue(), null, gold, "");
+                continue;
+            }
+            if (!e.getValue().getInputPinName().equals(testPIP.getInputPinName())) {
+                addDiff(DesignDiffType.SITEPIP_INPIN_NAME, e.getValue(), testPIP, gold, "");
+            }
+        }
+        for (Entry<BEL, SitePIP> e : testSitePIPs.entrySet()) {
+            addDiff(DesignDiffType.SITEPIP_EXTRA, null, e.getValue(), test, "");
+        }
+
+        return getDiffCount() - init;
+    }
+    
+    private Map<BEL, SitePIP> getSitePIPMap(SiteInst siteInst) {
+        Map<BEL, SitePIP> map = new HashMap<>();
+        for (SitePIP p : siteInst.getUsedSitePIPs()) {
+            SitePIP duplicate = map.put(p.getBEL(), p);
+            if (duplicate != null) {
+                System.out.println("WARNING: Multiple SitePIPs active on BEL " + p.getBELName() + " of Site "
+                        + siteInst.getSiteName());
+            }
+        }
+        return map;
+    }
+
+    private Map<String, PIP> getPIPMap(Net net) {
+        Map<String, PIP> map = new HashMap<>();
+        for (PIP p : net.getPIPs()) {
+            map.put(p.toString(), p);
+        }
+        return map;
+    }
+
+    /**
+     * Compares two nets for differences in routing (PIPs). Two flags can be set to
+     * affect its operation: {@link #setComparePIPs(boolean)} (default: true) and
+     * {@link #setComparePIPFlags(boolean)} (default: false).
+     * 
+     * @param gold The expected or reference net.
+     * @param test The net to be compared against the reference net.
+     * @return The number of differences encountered while comparing the provided
+     *         nets. This number also mutates the total diff count stored in the
+     *         class.
+     */
+    public int compareNets(Net gold, Net test) {
```

Comment:
Should `compareNets` also compare pins (`SitePinInst`s)? I don't think that's necessary for/affects the application I have in mind, but could be a useful feature for a general purpose diff.

Reply:
`SitePinInst` objects are not stored in a DCP, so for that reason, I don't want them adding noise to the diff.  This could be added later as an optional feature if it is needed.